e2e/page_objects/programCatalog.js

```diff
@@ -30,21 +37,79 @@ class ProgramCatalog {
     }
 
     countCourses() {
-        return this.card.count();
+        return this.cards.count();
     }
 
     getPlaceholderText() {
         return this.searchBar.getAttribute('placeholder');
     }
 
     typeIntoSearchBar(text) {
-        return this.searchBar.sendKeys(text);
+        this.searchBar.sendKeys(text);
+        return browser.sleep(3000);
     }
 
     clearSearchBar() {
-        this.searchBar.sendKeys(protractor.Key.chord(protractor.Key.CONTROL,"a"));
+        this.searchBar.sendKeys(protractor.Key.chord(protractor.Key.CONTROL, "a"));
         return this.searchBar.sendKeys(protractor.Key.BACK_SPACE);
     }
+
+    selectedFilters() {
+        return this.selectedFiltersField.getText();
+    }
+
+    getResultCounter() {
+        return this.resultCounter.getText().then(text => {
+            let regex = /Results? \((\d+)\)/;
+            let token = text.match(regex);
+            return +token[1];
+        });
+    }
+
+    openFilterDropdown(text) {
+        return this.dropdownSelectorisVisible.isVisible().then(value => {
```

Comment:
Please, extract the this.dropdownSelectorisVisible.isVisible() into a separate function and call it here.